tests/5.0/target_update/test_target_update_mapper_from_discontiguous.c

```diff
@@ -17,40 +16,58 @@
 
 #define N 1024
 
-typedef struct newvec {
+typedef struct{
   size_t len;
   double *data;
-} newvec_t;
+} T;
 
-size_t i;
+int i;
 int errors = 0;
 
-#pragma omp declare mapper(newvec_t v) map(v, v.data[0:v.len:2])
+#pragma omp declare mapper(S: T v) map(v, v.len, v.data[0:v.len])
 
 int target_update_from_mapper() {
   
-  newvec_t s; 
-
-  s.data = (double *)calloc(N,sizeof(double));
+  T s;
+  
   s.len = N;
-  for (i = 0; i< s.len; i++){ //initalize the struct
-    s.data[i] = 0;
-  }
-  #pragma omp target
-  {
-    for (i = 0; i < s.len; i += 2) {
+  s.data = (double *)calloc(N,sizeof(double));
+  
+  
+  #pragma omp target  
+  { 
+    for (int i = 0; i < s.len; i++) {
       s.data[i] = i;
     }
   }//end target
-  #pragma omp target update from(s)
-  for (i =0; i < s.len; i += 2) {
-    OMPVV_TEST_AND_SET(errors, s.data[i] != i);
+  
+  #pragma omp target data map(mapper(S), to:s) 
+  { 
+    #pragma omp target defaultmap(none) map(to:s.len)
```

Comment:
Do we need `map(to:s.len)` here? It seems to be redundant.

Reply:
est_update_mapper_dis.c:37:25: error: variable 's' must have explicitly specified data sharing attributes, data mapping attributes, or in an is_device_ptr clause
    for (int i = 0; i < s.len; i++) {       
                        ^